arroyo-connectors/src/polling_http.rs

```diff
@@ -0,0 +1,221 @@
+use std::convert::Infallible;
+
+use anyhow::anyhow;
+use arroyo_rpc::OperatorConfig;
+use arroyo_types::string_to_map;
+use axum::response::sse::Event;
+use reqwest::{Client, Request};
+use tokio::sync::mpsc::Sender;
+use typify::import_types;
+
+use arroyo_rpc::types::{ConnectionSchema, ConnectionType, TestSourceMessage};
+use serde::{Deserialize, Serialize};
+
+use crate::{construct_http_client, pull_opt, pull_option_to_i64, Connection, EmptyConfig};
+
+use super::Connector;
+
+const TABLE_SCHEMA: &str = include_str!("../../connector-schemas/polling_http/table.json");
+
+import_types!(schema = "../connector-schemas/polling_http/table.json");
+const ICON: &str = include_str!("../resources/http.svg");
+
+pub struct PollingHTTPConnector {}
+
+impl PollingHTTPConnector {
+    fn construct_test_request(
+        client: &Client,
+        config: &PollingHttpTable,
+    ) -> anyhow::Result<Request> {
+        let mut req = client.request(
+            match config.method {
+                None | Some(Method::Get) => reqwest::Method::GET,
+                Some(Method::Post) => reqwest::Method::POST,
+                Some(Method::Put) => reqwest::Method::PUT,
+                Some(Method::Patch) => reqwest::Method::PATCH,
```

Comment:
Why would a PATCH request be used? Or really anything other than GET?

Reply:
There are APIs that you need to POST to, and seemed worth not artificially limiting it